pandas/io/gbq.py

```diff
@@ -148,15 +164,62 @@ def get_credentials(self):
 
         return credentials
 
+    def get_service_account_credentials(self):
+        from oauth2client.client import SignedJwtAssertionCredentials
+        from os.path import isfile
+
+        try:
+            if isfile(self.private_key):
+                with open(self.private_key) as f:
+                    json_key = json.loads(f.read())
+            else:
+                # ugly hack: 'private_key' field has new lines inside, they break json parser, but we need to preserve them
+                json_key = json.loads(self.private_key.replace('\n', '   '))
+                json_key['private_key'] = json_key['private_key'].replace('   ', '\n')
+
+            return SignedJwtAssertionCredentials(
+                json_key['client_email'],
+                bytes(json_key['private_key'], 'UTF-8'),
```

Comment:
I'm receiving the following exception :

```
======================================================================
ERROR: test_should_be_able_to_make_a_connector (pandas.io.tests.test_gbq.TestGBQConnectorServiceAccountKeyPathIntegration)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tony/pandas-gbq_service_account/pandas/io/tests/test_gbq.py", line 167, in setUp
    self.sut = gbq.GbqConnector(PROJECT_ID, private_key=PRIVATE_KEY_JSON_PATH)
  File "/home/tony/pandas-gbq_service_account/pandas/io/gbq.py", line 128, in __init__
    self.credentials = self.get_credentials()
  File "/home/tony/pandas-gbq_service_account/pandas/io/gbq.py", line 145, in get_credentials
    return self.get_service_account_credentials()
  File "/home/tony/pandas-gbq_service_account/pandas/io/gbq.py", line 186, in get_service_account_credentials
    raise InvalidPrivateKeyFormat("Service account private key should be valid JSON (file path or string contents) "
InvalidPrivateKeyFormat: Service account private key should be valid JSON (file path or string contents) with at least two keys: 'client_email' and 'private_key'. Can be obtained from google developers console. 
```
If I remove the exception handling, I get the following error:
```
======================================================================
ERROR: test_should_be_able_to_make_a_connector (pandas.io.tests.test_gbq.TestGBQConnectorServiceAccountKeyPathIntegration)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tony/pandas-gbq_service_account/pandas/io/tests/test_gbq.py", line 167, in setUp
    self.sut = gbq.GbqConnector(PROJECT_ID, private_key=PRIVATE_KEY_JSON_PATH)
  File "/home/tony/pandas-gbq_service_account/pandas/io/gbq.py", line 128, in __init__
    self.credentials = self.get_credentials()
  File "/home/tony/pandas-gbq_service_account/pandas/io/gbq.py", line 145, in get_credentials
    return self.get_service_account_credentials()
  File "/home/tony/pandas-gbq_service_account/pandas/io/gbq.py", line 182, in get_service_account_credentials
    bytes(json_key['private_key'], 'UTF-8'),
TypeError: str() takes at most 1 argument (2 given)
```

The issue is resolved with the issue with the following code:
```
            json_key['private_key'] = json_key['private_key'].replace('   ', '\n')

            if compat.PY3:
                json_key['private_key'] = bytes(json_key['private_key'], 'UTF-8')

        return SignedJwtAssertionCredentials(
            json_key['client_email'],
            json_key['private_key'],
            self.scope,
        )
```


```
tony@tonypc:~/pandas-gbq_service_account/pandas/io/tests$ nosetests test_gbq.py:TestGBQConnectorServiceAccountKeyPathIntegration -v --nocapture
test_should_be_able_to_get_a_bigquery_service (pandas.io.tests.test_gbq.TestGBQConnectorServiceAccountKeyPathIntegration) ... ok
test_should_be_able_to_get_results_from_query (pandas.io.tests.test_gbq.TestGBQConnectorServiceAccountKeyPathIntegration) ... ok
test_should_be_able_to_get_schema_from_query (pandas.io.tests.test_gbq.TestGBQConnectorServiceAccountKeyPathIntegration) ... ok
test_should_be_able_to_get_valid_credentials (pandas.io.tests.test_gbq.TestGBQConnectorServiceAccountKeyPathIntegration) ... ok
test_should_be_able_to_make_a_connector (pandas.io.tests.test_gbq.TestGBQConnectorServiceAccountKeyPathIntegration) ... ok

----------------------------------------------------------------------
Ran 5 tests in 3.756s

OK
```